src/instana/collector/helpers/runtime.py

```diff
@@ -10,29 +11,115 @@
     return any(map(lambda p: PATH_OF_AUTOTRACE_WEBHOOK_SITEDIR in p, sys.path))
 
 
+def parse_service_pair(pair: str) -> List[str]:
+    """
+    Parses a pair string to prepare a list of ignored endpoints.
+
+    @param pair: String format:
+        - "service1:endpoint1,endpoint2" or "service1:endpoint1" or "service1"
+    @return: List of strings in format ["service1.endpoint1", "service1.endpoint2", "service2"]
+    """
+    pair_list = []
+    if ":" in pair:
+        service, endpoints = pair.split(":", 1)
+        service = service.strip()
+        endpoint_list = [ep.strip() for ep in endpoints.split(",") if ep.strip()]
+
+        for endpoint in endpoint_list:
+            pair_list.append(f"{service}.{endpoint}")
+    else:
+        pair_list.append(pair)
+    return pair_list
+
+
+def parse_ignored_endpoints_string(params: str) -> List[str]:
+    """
+    Parses a string to prepare a list of ignored endpoints.
+
+    @param params: String format:
+        - "service1:endpoint1,endpoint2;service2:endpoint3" or "service1;service2"
+    @return: List of strings in format ["service1.endpoint1", "service1.endpoint2", "service2"]
+    """
+    ignore_endpoints = []
+    if params:
+        service_pairs = params.split(";")
+
+        for pair in service_pairs:
+            if pair.strip():
+                ignore_endpoints += parse_service_pair(pair)
+    return ignore_endpoints
+
+
+def parse_ignored_endpoints_dict(params: Dict[str, Any]) -> List[str]:
+    """
+    Parses a dictionary to prepare a list of ignored endpoints.
+
+    @param params: Dict format:
+        - {"service1": ["endpoint1", "endpoint2"], "service2": ["endpoint3"]}
+    @return: List of strings in format ["service1.endpoint1", "service1.endpoint2", "service2"]
+    """
+    ignore_endpoints = []
+
+    for service, endpoints in params.items():
+        if not endpoints:  # filtering all service
+            ignore_endpoints.append(service)
+        else:  # filtering specific endpoints
+            for endpoint in endpoints:
+                ignore_endpoints.append(f"{service}.{endpoint}")
+
+    return ignore_endpoints
+
+
+def parse_ignored_endpoints(params: Union[Dict[str, Any], str]) -> List[str]:
+    """
+    Parses input to prepare a list for ignored endpoints.
+
+    @param params: Can be either:
+        - String: "service1:endpoint1,endpoint2;service2:endpoint3" or "service1;service2"
+        - Dict: {"service1": ["endpoint1", "endpoint2"], "service2": ["endpoint3"]}
+    @return: List of strings in format ["service1.endpoint1", "service1.endpoint2", "service2"]
+    """
+    try:
+        if isinstance(params, str):
+            return parse_ignored_endpoints_string(params)
+        elif isinstance(params, dict):
+            return parse_ignored_endpoints_dict(params)
+        else:
+            return []
+    except Exception as e:
+        logger.debug("Error parsing ignored endpoints: %s", str(e))
+        return []
+
```

Comment:
I think these methods are in the wrong place since they are not part of the `Collector` scope but the `Config` one.
It should be located at `src/instana/util/runtime.py` or, even better, in a new file called `src/instana/util/config.py`.